daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -80,6 +83,28 @@ func init() {
 	}
 }
 
+func getStreamName(ctx logger.Context) (name string, err error) {
+	if ctx.Config[logStreamKey] != "" {
+		name = ctx.Config[logStreamKey]
+	} else if ctx.Config[logStreamTemplateKey] != "" {
+		var t *template.Template
+		t, err = template.New("t").Parse(ctx.Config[logStreamTemplateKey])
+		if err == nil {
+			buff := bytes.NewBuffer(nil)
+			err = t.Execute(buff, ctx)
+			if err == nil {
+				name = buff.String()
+			}
+		}
+	} else {
+		name = ctx.ContainerID
+	}
+	if strings.IndexAny(name, ":*") >= 0 {
```

Comment:
I was using the logStreamName 'pattern' regexp: "[^:*]*" -- which excludes ':'s and '*'s unless I'm terribly misreading - wouldn't be hard to test to find out which statement in the same doc is true